File: countnum.py
def factorial(n):
    if n >1:
        return n *factorial(n-1)
num =int(input())    
for i in range(num+1):
    count=0
    str=input()
    k=input()
    for j in range(len(str)):
        if str[j]==k:
            count=count+1
    print(factorial(count))        

File: countnum.c
#include <stdio.h>

unsigned long factorial(int n) {
    if (n == 0 || n == 1)
        return 1;
    else if(n>1)
        return (n * factorial(n - 1)+1000000007)%1000000007;
}

int countOccurrences(char* str, char c) {
    int count = 0;
    while (*str) {
        if (*str == c)
            count++;
        str++;
    }
    return count;
}

int main() {
    int numTestCases;
    scanf("%d", &numTestCases);

    while (numTestCases > 0) {
        char inputString[100000];
        scanf("%s", inputString);

        int targetNumber;
        scanf("%d", &targetNumber);

        int count = countOccurrences(inputString, targetNumber + '0');
        

        printf("%lu ", factorial(count));

        numTestCases--;
    }

    return 0;
}
